patches/minecraft/net/minecraft/block/state/IBlockProperties.java.patch

```diff
@@ -24,6 +24,7 @@
 +    //Forge added functions
 +    boolean doesSideBlockRendering(IBlockAccess world, BlockPos pos, EnumFacing side);
 +    boolean isSideSolid(IBlockAccess world, BlockPos pos, EnumFacing side);
++    boolean doesSideBlockChestOpening(IBlockAccess world, BlockPos pos, EnumFacing side);
```

Comment:
probably worth taking advantage of default methods, return this.getBlock().doesSideBlockChestOpening(this....)

Reply:
The problem is that `getBlock` is defined by `IBlockState`, not `IBlockProperties`

Reply:
You can add a default method in `IBlockState` then.

Reply:
Sure, that'll lighten the patch a bit.

Reply:
Ok I took a closer look and we don't even patch `IBlockState`, and `IBlockProperties` has no access to the block. I could put it all into a default method in `IBlockState` but I think it's probably best to leave it the way it is since the other forge patches so far work this way.

Reply:
Or we can wait until an `AbstractMethodError` is thrown (not likely since the old users may not call this new method, and old compiled code should be still able to run)

Reply:
So the error is here! #4603 happens! Also see sp614x/optifine#1071